src/chains/tests/test_views.py

```diff
@@ -252,3 +219,69 @@ def test_null_ens_registry_address(self):
 
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.data["ens_registry_address"], None)
+
+
+class ChainGasPriceTests(APITestCase):
+    faker = Faker()
+
+    def test_oracle_json_payload_format(self):
+        chain = ChainFactory.create(
+            id=1, gas_price_oracle_url=self.faker.url(), gas_price_fixed_wei=None
+        )
+        url = reverse("v1:chains:detail", args=[1])
+        expected_oracle_json_payload = {
+            "type": "oracle",
+            "url": chain.gas_price_oracle_url,
+            "gasParameter": chain.gas_price_oracle_parameter,
+            "gweiFactor": "{0:.9f}".format(chain.gas_price_oracle_gwei_factor),
+        }
+
+        response = self.client.get(path=url, data=None, format="json")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json()["gasPrice"], expected_oracle_json_payload)
+
+    def test_fixed_gas_price_json_payload_format(self):
+        chain = ChainFactory.create(id=1, gas_price_fixed_wei=self.faker.pyint())
+        url = reverse("v1:chains:detail", args=[1])
+        expected_oracle_json_payload = {
+            "type": "fixed",
+            "weiValue": str(chain.gas_price_fixed_wei),
+        }
+
+        response = self.client.get(path=url, data=None, format="json")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json()["gasPrice"], expected_oracle_json_payload)
+
+    def test_oracle_with_fixed(self):
+        chain = ChainFactory.create(
+            id=1,
+            gas_price_oracle_url=self.faker.url(),
+            gas_price_fixed_wei=self.faker.pyint(),
+        )
+        url = reverse("v1:chains:detail", args=[1])
+        expected_error_body = {
+            "detail": f"The gas price oracle or a fixed gas price was not provided for chain {chain.id}"
+        }
+
+        response = self.client.get(path=url, data=None, format="json")
+
+        self.assertEqual(response.status_code, 500)
+        self.assertEqual(response.json(), expected_error_body)
+
+    def test_fixed_gas_256_bit(self):
+        ChainFactory.create(
+            id=1,
+            gas_price_fixed_wei="115792089237316195423570985008687907853269984665640564039457584007913129639935",
+        )
+        url = reverse("v1:chains:detail", args=[1])
+        expected_oracle_json_payload = {
+            "type": "fixed",
+            "weiValue": "115792089237316195423570985008687907853269984665640564039457584007913129639935",
```

Comment:
Ok cool I think I answered my own question above :D

Reply:
You can see the payload example in the PR description: https://github.com/gnosis/safe-config-service/pull/157#issue-689243974

This is what the CGW should expect.